### PR TITLE
Fix name length and symbol issue

### LINKED ISSUE #138: Name allows numbers and symbols in them

### PR DESCRIPTION
**Pull request by bingmybongg**
Fixes #138, #169

### CODE REVIEW

File: src/test/java/seedu/address/model/person/NameTest.java [hunk 1]
```diff
@@ -24,18 +24,33 @@ public void isValidName() {
...
-        assertFalse(Name.isValidName("peter*")); // contains non-alphanumeric characters
+        assertFalse(Name.isValidName("^")); // non-alphabetic character
+        assertFalse(Name.isValidName("peter*")); // contains symbol
+        assertFalse(Name.isValidName("12345")); // numbers only (no longer valid)
+        assertFalse(Name.isValidName("peter the 2nd")); // contains number
+        assertFalse(Name.isValidName("-John")); // starts with hyphen
+        assertFalse(Name.isValidName("'John")); // starts with apostrophe
+        assertFalse(Name.isValidName(" John")); // starts with space
+        assertFalse(Name.isValidName("A" + "a".repeat(100))); // 101 characters (too long)
+        assertFalse(Name.isValidName("A" + "A".repeat(100))); // 101 characters (too long)
```
Comment (momentumnn) [inline src/test/java/seedu/address/model/person/NameTest.java]: Do we need 4 instances of the same boundary case?
